unit-tests/shared/src/test/scala/javalib/lang/StringBufferTest.scala

```diff
@@ -35,6 +35,11 @@ class StringBufferTest {
     assertEquals("2.5", newBuf.append(2.5f).toString)
     assertEquals("3.5", newBuf.append(3.5).toString)
   }
+  
+  @Test def appendFloats(): Unit = {
```

Comment:
Most importantly, having this test is an improvement.

Naming thing is always hard and a matter of personal preference, intent, & experience.

The scope of discussion being "floating point numbers" and
that having different, defined, and sometimes overloaded  usages, the title `appendFloats()` leads to be expect 
a series of "f" (IEEE float) appends, with not explicit or implied "d" (IEEE double).


Since this uses, probably by design, both "f" and "d" floating
point, perhaps something like "testMixedFloatingPointAppends()`.  You know your intent, so you may be able to come up with a better name (testMixedIEEE754Appends()?"

Reply:
I just noticed that this test is in `StringBufferTest`. 

It will test this PR there, but, perhaps provincially, I would have expected
it in (and had assumed, reading more quickly than a reviewer should...)
that it would be in `StringBuilder` (as `StringBuilder` is the more current
idiom).

Reply:
>The scope of discussion being "floating point numbers...

Actually I was also disturbed by the naming of `def appendFloat()`, function that is doing stuffs with both Float and Double values, and which is just above the added test.
I stupidly decided to stick to the chosen naming, but this could be improved, your are right.

>I just noticed that this test is in `StringBufferTest`.

Totally right, I should have added this test in both files, my mistake.

Reply:
> I stupidly 

Given the complexity of all of this tangle, following precedent is not stupid at all.  
It is a good starting point, up until getting to the "Does what they were doing suit
present circumstances/needs point."

Reply:
I think that testing both Float and Double is an improvement.  

Given a finite personal & professional lifespan, I personally would put one set of tests.
into `StringBuilder` with a comment that they exercise the method(s) inherited from
`AbstractStringBuilder` .     

The older, not quite deprecated, `StringBuffer` could be left with either a deletion (i.e. nothing visible)
a comment that `StringBuffer` inherits those methods from `AbstractStringBuilder` and they
are exercised in `StringBuilder`.

Normally, I would go the comment route in `StringBuffer`, but here I think it invites confusion.

Reply:
If it's fine to you, I would go for the unit tests implemented in last commit (https://github.com/scala-native/scala-native/pull/2917/commits/7983512fe684f2bfb05e023df0c5e24b8a802dc5)